cmake/CMakeDefinitions.cmake

```diff
@@ -92,10 +77,17 @@ endif()
 include(CheckIncludeFileCXX)
 CHECK_INCLUDE_FILE_CXX(span HAS_STD_SPAN_HEADER)
 
-if(HAS_STD_SPAN_HEADER AND CMAKE_CXX_STANDARD GREATER_EQUAL 20)
+if(SPHERAL_UNIFIED_MEMORY)
```

Comment:
The choice of which span (std::span vs boost::span) is a separate question from whether or not we're using SPHERAL_UNIFIED_MEMORY, so I don't this this check should be restricted to that case.